doc/NAC/PnacSonicHLD.md

```diff
@@ -0,0 +1,371 @@
+            },            
+            "queue1_group1": {            
+                    "trap_action":"trap",                    
+                    "trap_priority":"1",                    
+                    "queue": "1",                    
+                    "meter_type":"packets",                    
+                    "mode":"sr_tcm",                    
+                    "cir":"6000",                    
+                    "cbs":"6000",                    
+                    "red_action":"drop"                    
+            },
+            "queue1_group2": {            
+                    "trap_action":"trap",                    
+                    "trap_priority":"1",                    
+                    "queue": "1",                    
+                    "meter_type":"packets",                    
+                    "mode":"sr_tcm",                    
+                    "cir":"600",                    
+                    "cbs":"600",                    
+                    "red_action":"drop"                    
+            },
+            "queue2_group1": {            
+                    "cbs": "1000",                    
+                    "cir": "1000",                    
+                    "genetlink_mcgrp_name": "packets",                    
+                    "genetlink_name": "psample",                    
+                    "meter_type": "packets",                    
+                    "mode": "sr_tcm",                    
+                    "queue": "2",                    
+                    "red_action": "drop",                    
+                    "trap_action": "trap",                    
+                    "trap_priority": "1"                     
+             }             
+    },    
+    "COPP_TRAP": {    
+            "nac": {            
+                    "trap_ids": "eapol",                    
+                    "trap_group": "queue4_group3"                    
+            }            
+    }    
+}
+```
+After configurating “***NAC Adminstate***” to “***up***” for a given interface, update in config_db triggers 1following actions
+Update learn_mode for the specific interface in config_db to “***SAI_BRIDGE_PORT_FDB_LEARNING_MODE_DROP***”
+Flush FDB for the bridge that’s part of the interface
+Enable NAC for the interface maintained by hostapd. This point onwards hostapd starts listening to EAPOL packets received on this interface.
+
+After configurating “***NAC Adminstate***” to “down” for a given interface, update in config_db triggers following actions,
+Action to disable NAC for the interface maintained by hostapd. This point onwards hostapd stops listening to EAPOL packets received on this interface.
+Update learn_mode for the specific interface in config_db to “***SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW***”
+### 3.2.4 Actions performed after receiving state change notification from hostapd
```

Comment:
Can you add details w.r.t how this is integrated with the upper layer protocols LACP, STP etc?

Reply:
For this HLD version only port based Authentication/Authorisation is supported. 
Since STP is not yet merged with mainstream, I have not considered impact to STP, as we do not have mechanism to share port status with STP. As when STP feature gets implemented in main branch, the design will be updated.

Reply:
Answered both queries on 20-02-2023.